pkg/controller/user-cluster-controller-manager/resources/resources/mla/monitoring-agent/configmap.go

```diff
@@ -0,0 +1,325 @@
+/*
+Copyright 2021 The Kubermatic Kubernetes Platform contributors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package monitoringagent
+
+import (
+	"bytes"
+	"text/template"
+
+	"github.com/Masterminds/sprig/v3"
+
+	"k8c.io/kubermatic/v2/pkg/resources"
+	"k8c.io/kubermatic/v2/pkg/resources/reconciling"
+
+	corev1 "k8s.io/api/core/v1"
+)
+
+type Config struct {
+	MLAGatewayURL       string
+	TLSCertFile         string
+	TLSKeyFile          string
+	TLSCACertFile       string
+	CustomScrapeConfigs string
+	HAClusterIdentifier string
+}
+
+func ConfigMapCreator(config Config) reconciling.NamedConfigMapCreatorGetter {
+	return func() (string, reconciling.ConfigMapCreator) {
+		return resources.UserClusterMonitoringAgentConfigMapName, func(configMap *corev1.ConfigMap) (*corev1.ConfigMap, error) {
+			if configMap.Data == nil {
+				configMap.Data = map[string]string{}
+			}
+			t, err := template.New("prometheus").Funcs(sprig.TxtFuncMap()).Parse(configTemplate)
```

Comment:
What about changing the template name (`"prometheus"`) and filename in the configMap data (`"prometheus.yaml"`) to something non-prometheus as well?

Reply:
absolutely, missed it!